packages/qwik-city/runtime/src/link-component.tsx

```diff
@@ -1,75 +1,111 @@
+    ...linkProps
+  } = (() => props)();
   const clientNavPath = untrack(() => getClientNavPath({ ...linkProps, reload }, loc));
-  const prefetchDataset = untrack(() => getPrefetchDataset(props, clientNavPath, loc));
   linkProps['preventdefault:click'] = !!clientNavPath;
   linkProps.href = clientNavPath || originalHref;
-  const onPrefetch =
-    prefetchDataset != null
-      ? event$((ev: any, elm: HTMLAnchorElement) =>
-          prefetchLinkResources(elm as HTMLAnchorElement, ev.type === 'qvisible')
-        )
-      : undefined;
-  const handleClick = event$(async (_: any, elm: HTMLAnchorElement) => {
-    if (!elm.hasAttribute('preventdefault:click')) {
-      // Do not enter the nav pipeline if this is not a clientNavPath.
-      return;
-    }
 
-    if (elm.hasAttribute('q:nbs')) {
-      // Allow bootstrapping into useNavigate.
-      await nav(location.href, { type: 'popstate' });
-    } else if (elm.href) {
-      elm.setAttribute('aria-pressed', 'true');
-      await nav(elm.href, { forceReload: reload, replaceState, scroll });
-      elm.removeAttribute('aria-pressed');
-    }
-  });
+  const prefetchData = untrack(
+    () =>
+      (!!clientNavPath &&
+        prefetchProp !== false &&
+        prefetchProp !== 'js' &&
+        shouldPrefetchData(clientNavPath, loc)) ||
+      undefined
+  );
+
+  const prefetch = untrack(
+    () =>
+      prefetchData ||
+      (!!clientNavPath && prefetchProp !== false && shouldPrefetchSymbols(clientNavPath, loc))
+  );
+
+  const handlePrefetch = prefetch
+    ? event$((_: any, elm: HTMLAnchorElement) => {
+        if ((navigator as any).connection && (navigator as any).connection.saveData) {
+          return;
+        }
+
+        if (elm && elm.href) {
+          const url = new URL(elm.href);
+          prefetchSymbols(url.pathname);
+
+          if (elm.hasAttribute('data-prefetch')) {
+            loadClientData(url, elm, { prefetchSymbols: false });
+          }
+        }
+      })
+    : undefined;
+
+  const handleClick = clientNavPath
+    ? event$(async (event: any, elm: HTMLAnchorElement) => {
+        if (!(event as PointerEvent).defaultPrevented) {
+          // Don't enter the nav pipeline if default hasn't been prevented.
+          return;
+        }
+
+        if (elm.hasAttribute('q:nbs')) {
+          // Allow bootstrapping into useNavigate.
+          await nav(location.href, { type: 'popstate' });
+        } else if (elm.href) {
+          await nav(elm.href, { forceReload: reload, replaceState, scroll });
+        }
+      })
+    : undefined;
+
   return (
     <a
       {...linkProps}
       onClick$={[onClick$, handleClick]}
-      data-prefetch={prefetchDataset}
-      onMouseOver$={onPrefetch}
-      onFocus$={onPrefetch}
-      onQVisible$={onPrefetch}
+      data-prefetch={prefetchData}
+      onMouseOver$={isDev ? handlePrefetch : undefined}
+      onFocus$={isDev ? handlePrefetch : undefined}
+      onQVisible$={!isDev ? handlePrefetch : undefined}
```

Comment:
@jordanw66 actually, shouldn't these include the given props if any (if so, that's an existing bug)?

And I don't understand why the handling happens only when isDev?